### PR TITLE
Fix issue #301: Replace raw pointer usage with std::unique_ptr

### PR DESCRIPTION
**Pull request by Glebanister**
- Create helper function Detail::makeUnique for creating smart pointers
- Replace raw pointers with smart std::unique_ptr

### CODE REVIEW

**Comment by pjotrp** [comment]
Love it!

File: src/genotypeSummary.cpp [hunk 1]
```diff
@@ -327,19 +328,21 @@ int main(int argc, char** argv) {
...
-	genotype * populationTarget      ;
+  using Detail::makeUnique;
+
+	unique_ptr<genotype> populationTarget      ;
 
 	if(type == "PL"){
-	  populationTarget     = new pl();
+	  populationTarget     = makeUnique<pl>();
 	}
 	if(type == "GL"){
```
Comment (elfring) [inline src/genotypeSummary.cpp:338]: I suggest to add also else branches for shown type checks.

File: src/hapLrt.cpp [hunk 2]
```diff
@@ -593,40 +596,24 @@ int main(int argc, char** argv) {
...
-
-	delete populationTarget;
-	delete populationBackground;
-	delete populationTotal;
+	loadPhased(haplotypes, populationTotal.get(), nsamples);
 
-	populationTarget     = NULL;
-	populationBackground = NULL;
-	populationTotal      = NULL;
+	;
```
Comment (elfring) [inline src/hapLrt.cpp:611]: I find the left over three lines questionable.

Comment (Glebanister) [reply src/hapLrt.cpp:611]: I believe that assigning these pointers to `NULL` was necessary. Previously, memory by pointer was deleted manually and then they were assigned `NULL` for some purpose, even though they continue and are not used. The destructor `unique_ptr` simply frees memory now.

Comment (elfring) [reply src/hapLrt.cpp:611]: How do you think about to avoid empty statements in such source code?

**Pull request merged**